### DISCUSSION TITLE
Are there any GUIs for AVA

### DISCUSSION THREAD

**Discussion by jakobrosenberg** [avajs/ava#2747] [Q&A] [answered]
Is there a GUI tool that integrates with AVA and lets you filter which tests to run in realtime?

**Comment by sindresorhus** [comment, marked as answer]
I'm not aware of any GUI. The closest you'll get is [watch mode](https://github.com/avajs/ava/blob/main/docs/recipes/watch-mode.md), which will only run changed files. Ideally, watch mode would support interactively filtering tests, but that's not supported yet: https://github.com/avajs/ava/issues/1530.

**Reply by jakobrosenberg** [reply]
Thanks for the quick reply, Sindre. Maybe someday.

**Comment by KillyMXI** [comment]
[Testify](https://marketplace.visualstudio.com/items?itemName=felixjb.testify) for VScode does work. Albeit it's only represents itself in a code lens.

It's a shame all Test Explorer UI AVA plugins are either broken or dodgy.
